### PR TITLE
MDP-204: Utility getDateByTimestamp

### PR DESCRIPTION
**Pull request by Stephaaan**
utility that takes a timestamp and returns formatted representation of a day ("Dnes 1.1", "Zajtra 2.1", "Streda 3.1"

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -58,6 +58,7 @@
     "@emotion/styled": "^10.0.10",
     "@reach/router": "^1.2.1",
     "classnames": "^2.2.6",
+    "dayjs": "^1.8.14",
```
Comment (pggl) [inline package.json:62]: add types for this

File: src/utils/date.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const diff = dayjs()
+    .startOf("day")
+    .diff(dayjs(timestamp * 1000).startOf("day"), "day", true);
+
+  const day =
+    diff <= -2 ? null : diff <= -1 ? "Zajtra" : diff <= 0 ? "Dnes" : null;
+  if (day) {
+    return (
+      day +
+      dayjs(timestamp * 1000)
```
Comment (pggl) [inline src/utils/date.ts]: create function for this

File: src/utils/date.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import dayjs from "dayjs";
+import "dayjs/locale/sk";
+
+export const getDateByTimestamp = (timestamp: number): string => {
+  const diff = dayjs()
```
Comment (pggl) [inline src/utils/date.ts]: extract this into separate function

File: src/utils/date.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    .startOf("day")
+    .diff(dayjs(timestamp * 1000).startOf("day"), "day", true);
+
+  const day =
+    diff <= -2 ? null : diff <= -1 ? "Zajtra" : diff <= 0 ? "Dnes" : null;
```
Comment (pggl) [inline src/utils/date.ts]: create function for this and use defensive style for this

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/utils/date.test.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import { getDateByTimeString, getDifferenceBetweenDate } from "./date";
+
+test("Testing difference between dates - no difference", () => {
+  expect(getDifferenceBetweenDate(new Date("2019-05-16 00:00:00"))).toEqual(0);
+});
+
```
Comment (garySiladi) [inline src/utils/date.test.ts:13]: can we have unified lining? either have empty line after tests or not. I would prefer having an empty line between every test

**Review by garySiladi — CHANGES_REQUESTED** [review]

File: src/utils/date.test.ts [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+test("Convert date to text - today", () => {
+  expect(getDateByTimeString("2019-05-16 00:00:00")).toEqual(
+    "Dnes - Štvrtok 16. máj"
+  );
+});
```
Comment (pggl) [inline src/utils/date.test.ts]: add to midnight

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/utils/date.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import dayjs from "dayjs";
+import "dayjs/locale/sk";
+
+// tested
```
Comment (pggl) [inline src/utils/date.ts]: remove comments

File: src/utils/date.ts [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+export const getCalendar = (difference: number): "Dnes" | "Zajtra" | null => {
+  const roundedDifference = Math.floor(difference);
+  if (roundedDifference < -1 || roundedDifference > 0) {
```
Comment (pggl) [inline src/utils/date.ts]: use constant for numbers

File: src/utils/date.ts [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    return null;
+  }
+  return roundedDifference === -1 ? "Zajtra" : "Dnes";
+};
+
+// tested
+export const getDifferenceBetweenDate = (dateObject: Date): number => {
+  return dayjs()
```
Comment (pggl) [inline src/utils/date.ts]: reverse

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/utils/date.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
+import dayjs from "dayjs";
+import "dayjs/locale/sk";
+const TODAY = 0;
+const TOMMOROW = 1;
```
Comment (garySiladi) [inline src/utils/date.ts]: if we don't need TODAY and TOMORROW in other functions, we could declare them in the `getCalendar` function

Comment (garySiladi) [reply src/utils/date.ts]: also can you put an empty line before the `getCalendar` function

Comment (Stephaaan) [reply src/utils/date.ts]: Peter's decision to put the constants before function declaration, but I'll move them to the function

File: src/utils/date.ts [hunk 11]
```diff
@@ -0,0 +1,36 @@
+import dayjs from "dayjs";
+import "dayjs/locale/sk";
+const TODAY = 0;
```
Comment (pggl) [inline src/utils/date.ts]: add empty line

**Review by pggl — CHANGES_REQUESTED** [review]

**Comment by Stephaaan** [comment]
Can be the text which is compared by test with the value, that function returns hardcoded? Because if I write a function that generates me that text by actual date, (I mean text "Dnes - 18. apríl"), I have to write tests also for that function with hardcoded values. In this version, I need to rewrite the right return values every day (because every day is date changing :D) So, Do I have to write a function that generates me this texts by date or just let it like this and rewrite tests in future if necessary?

File: src/utils/date.ts [hunk 12]
```diff
@@ -0,0 +1,37 @@
+import dayjs from "dayjs";
+import "dayjs/locale/sk";
+
+export const getCalendar = (difference: number): "Dnes" | "Zajtra" | null => {
```
Comment (garySiladi) [inline src/utils/date.ts]: I would rename this function to `getTodayOrTomorrow`

File: src/utils/date.test.ts [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+
+beforeAll(() => {
+  MockDate.set("2019-05-18");
+});
+
+afterAll(() => {
+  MockDate.reset();
+});
+
+describe("Testing date utility", () => {
```
Comment (garySiladi) [inline src/utils/date.test.ts:18]: please re-check the linings after every test block. either have empty line after the blocks or not

**Review by garySiladi — CHANGES_REQUESTED** [review]

File: src/utils/date.ts [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+  const TODAY = 0;
+  const TOMMOROW = 1;
```
Comment (pggl) [inline src/utils/date.ts]: use enum

File: src/utils/date.test.ts [hunk 15]
```diff
@@ -0,0 +1,132 @@
...
+      expect(getFormattedDate(new Date("2019-05-21 00:00:00"))).toEqual(
+        "Utorok 21. máj"
+      );
+    });
+    test("Wednesday", () => {
+      expect(getFormattedDate(new Date("2019-05-22 00:00:00"))).toEqual(
+        "Streda 22. máj"
+      );
+    });
+    test("Thursday", () => {
```
Comment (pggl) [inline src/utils/date.test.ts]: use better description

**Review by pggl — CHANGES_REQUESTED** [review]

File: src/utils/date.test.ts [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+  MockDate.set("2019-05-18");
+});
+
+afterAll(() => {
+  MockDate.reset();
+});
+
+describe("Testing date utility", () => {
+  describe("Function that returns difference between actual date and date given as a parameter", () => {
+    test("Testing difference between dates - no difference", () => {
```
Comment (pggl) [inline src/utils/date.test.ts]: "Testing" is redundant

**Review by pggl — CHANGES_REQUESTED** [review]

**Review by pggl — APPROVED** [review]

**Review by garySiladi — APPROVED** [review]

**Pull request merged**